boards: st: stm32h573i_dk: Add TF-M non-secure app support

tpambor opened the pull request:
Add support for building Trusted Firmware-M (TF-M) non-secure applications for the STM32H573I-DK board.

tpambor commented:
Currently, fails to build with Zephyr SDK used in main due to a internal compiler error, https://github.com/zephyrproject-rtos/sdk-ng/issues/558. This is fixed starting GCC 13.

dsseng reviewed (commented):
Thanks, could you please consider these minor documentation fixups?

etienne-lms reviewed (commented):
Many CI build test failures (if not all) are due to a GCC 11/GCC 12 issue triggered in STM32 HAL for this SoC. It's been addressed for STM32U5 in TF-M  [1] first then solve in the STM32U5 HAL. The equivalent fix for STM32WBA65 was merged in TF-M [2] and should be addressed in the next release of STM32WBA65 HAL.

I see the issue needs to be addressed in TF-M also for STM32H573. It should be adressed in that SoC next HAL release but until then (and TF-M syncs on), it would need to be addressed in TF-M too, on the model of [1] and [2]. Do you want to post a change in TF-M mainline?

[1] https://github.com/TrustedFirmware-M/trusted-firmware-m/commit/578a6f485e085abd8f2f65d7762693fdac3237cd
[2] https://github.com/TrustedFirmware-M/trusted-firmware-m/commit/ec61e691a08bd21d15437ee2cff19f1d4688d5ee

erwango commented on boards/st/stm32h573i_dk/doc/index.rst:
```diff
@@ -149,6 +149,47 @@ Supported Features
++=========================================+==============================================================+
+| stm32h573i_dk                           | For building Trust Zone Disabled firmware for internal flash |
++-----------------------------------------+--------------------------------------------------------------+
+| stm32h573i_dk/stm32h573xx/ext_flash_app | For building Trust Zone Disabled firmware for external flash |
```
`with TrustZone disabled` more accurate than `without TrustZone`

tpambor commented on boards/st/stm32h573i_dk/stm32h573i_dk_stm32h573xx_ns.yaml:
```diff
@@ -0,0 +1,30 @@
+toolchain:
+  - zephyr
+ram: 256
+flash: 2048
```
Right, I also subtracted the 1kB for MCUboot header as in `ext_flash_app`.

tpambor committed:
boards: st: stm32h573i_dk: Add TF-M non-secure app support

Add support for building Trusted Firmware-M (TF-M) non-secure
applications for the STM32H573I-DK board.

Signed-off-by: Tim Pambor <tim.pambor@codewrights.de>
